iceberg/patched-iceberg-core/pom.xml

```diff
@@ -76,6 +76,8 @@
                   <outputDirectory>${project.build.directory}/classes</outputDirectory>
                   <excludes>
                       **/HadoopInputFile.class
+                      **/TableProperties.class
+                      **/ConfigProperties.class
```

Comment:
@zsmiskolczi  please correct me if i am wrong.

Patched-iceberg-core package need to have the patched TableProperties.class and ConfigProperties.class files mentioned here in it.
Currently in this PR, we are missing the patched files... Is there a different PR for these patched files?

Reply:
No. The original plan was to upgrade to 1.2.1 and additionally, some extra commits. As the ticket says: https://issues.apache.org/jira/browse/HIVE-27306

```
Extra that we have to take care of:

c3232b664745ebf761b6a74f4c5b55cc48bfd209: Hive: Use EnvironmentContext instead of Hive Locks to provide transactional commits after HIVE-26882;
```

So that commit was after 1.2.1 release but before 1.3.0. That means, when I ported it to Hive, I had to introduce those files in the patches to make this commit working. As you can see in the Iceberg commit (https://github.com/apache/iceberg/commit/c3232b664745ebf761b6a74f4c5b55cc48bfd209), those files are in the core module and it is not part of the Hive-Iceberg integration. But as at that time Iceberg dependency was 1.2.1, that was the way to see those changes in Iceberg. 

Right after I finished with the 1.2.1, 1.3.0 released. And it contained only few minor changes that haven't ported yet so I decided to port the remaining part in this PR. That is why this PR is not about upgrading from 1.1.0 to 1.2.1 but to 1.3.0. 

But, as 1.3.0 dependency contains those changes, with upgrading Iceberg dependency to 1.3.0, they could be removed (https://github.com/apache/hive/pull/4291/commits/ddba7deeb42ba8974e1cd49be96052da3ea259f4).

That is the reason why this PR doesn't contain those files.

Reply:
@simhadri-g , can I resolve that conversation?

Reply:
Small query here: 

> But, as 1.3.0 dependency contains those changes, with upgrading Iceberg dependency to 1.3.0, they could be removed (https://github.com/apache/hive/commit/ddba7deeb42ba8974e1cd49be96052da3ea259f4). 


Since these files are already in iceberg 1.3, shouldn’t we remove these two lines in the pom.xml ?

Reply:
Thank you for spotting this. I did some clean up and removed `SerializableTable` as well.

Reply:
`SerializableTable` in patched-iceberg-core had another method -> IncrementalAppendScan which was not present in upstream iceberg. 

Can we please check and restore the SerializableTable as needed? 


![image](https://github.com/apache/hive/assets/26441677/89a7c068-3126-4717-8883-ee10b71ddb19)